### PR TITLE
Fixed integration between autolink and autoembed

### LINKED ISSUE #2340: Autolink typing support breaks integration with autoembed plugin
## Type of report

Bug

## Provide description of the task

On major branch tests are failing:
`tests/plugins/autoembed`

screen shot 2018-08-27 at 15 22 55
screen shot 2018-08-27 at 15 25 58
screen shot 2018-08-27 at 15 26 07
screen shot 2018-08-27 at 15 26 12

## Other details

* Browser: any
* OS: …
* CKEditor version: …
* Installed CKEditor plugins: …

### PR DESCRIPTION
**Pull request by jacekbogdanski**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

Restored the old autolink behavior on paste event so it integrates with autoembed plugin smoothly.

Closes #2340

### CODE REVIEW

**Comment by jacekbogdanski** [comment]
TBO I find out changes in proposed PR superior to the previous implementation. I didn't like this little hacking with `afterPaste` event and it looks more readable. 

Also, I think it may be better to pass data through events instead of immediately inserting results. It gives us a potential possibility to modify passed data by different plugins before inserting it into the editor - which seems cleaner and more performant / less error-prone than fighting with already inserted content.

It doesn't mean that `autoembed` plugin doesn't require some refactoring e.g. mentioned dropping `autolink` dependency. Although I propose to extract it into a separate issue.

File: tests/plugins/autolink/autolink.js [hunk 1]
```diff
@@ -1,228 +1,279 @@
+'use strict';
```
[the remark this answers is not in the document]
Comment (jacekbogdanski) [reply tests/plugins/autolink/autolink.js]: I just restored old tests for `autolink` and not noticed it.

**Comment by jacekbogdanski** [comment]
I reused autoembed manual tests by attaching ticket numbers and editor version. However, it looks like `tests/plugins/autoembed/manual/notifications` is failing for non-embeddable links. It fails also on `master`, could you confirm it, so I can create separate ticket for it?

**Comment by Comandeer** [comment]
The mentioned test is failing only for `http://ckeditor.com/` as iframely introduced support for embedding every page with OpenGraph metadata. The second link, `http://foo/bar/`, however is still working correctly (so not working in this case ;)).

It shouldn't be fixed in this PR, but it's just replacing one link with another, so I think nothing wrong will happen if we do it here.

**Comment by jacekbogdanski** [comment]
On IE8 element names are uppercased when creating an element (upstream).

**Pull request merged**